# Fundamentals of Risk and Returns

## Simple Price Return

$$\large R_{t,t+1} = \frac{P_{t+1} - P_t}{P_t}$$

- The simple price return on the asset is nothing more than the price difference between two periods.

## Total Return: Price Return with Dividend

$$\large R_{t,t+1} = \frac{ P_{t+1} + D_{t,t+1} }{P_t} - 1 = \frac{ P_{t+1} + D_{t,t+1} - P_t }{P_t}$$

- The total return includes dividend payments for the holding period.

## Multi Period Returns

$$\large R_{t,t+2} = (1 + R_{t,t+1})(1+R_{t+1,t+2}) - 1$$

**Example**: You buy a stock that returns 10% on the first day and returns -3% on the second day.


$$R_{t,t+2} = (1 + 0.10)(1 - 0.03) - 1 = 0.067 = 6.70\%$$

## Annualizing Returns

**Annualized Returns** is the return you would get if the return that you are looking at has continued for a year. The return over the month is 1%. The annualized return would be $((1+0.01)^{12} - 1) = 0.12683 = 12.68\%$.

## Volatility - Standard Deviation and Variance

We compute the average of the square of the deviations from the mean. This measure is called Variance and is computed as follows.
$$\large \sigma^2_{R} = \frac{1}{N}\Sigma_{i=1}^{N} (R_i - \bar{R})^2$$ where $\bar{R}$ is the arithmetic mean of the returns.

Variance measures how much each return deviates from the mean of the returns. In other words, Variance represents the average of deviations.

If we take the square root of Variance, we get **Standard Deviation** and is computed as follows.

$$\large \sigma_{R} = \sqrt{\frac{1}{N}\Sigma_{i=1}^{N} (R_i - \bar{R})^2}$$ where $\bar{R}$ is the arithmetic mean of the returns.

In [73]:
returns.std()

Close    0.014428
dtype: float64

In [77]:
deviations = returns - returns.mean()
squared_deviations = deviations**2
variance = squared_deviations.mean()
volatility = np.sqrt(variance) # variance**0.5
volatility

Close    0.014043
dtype: float64

The two values don't match exactly. `std()` method uses $N-1$ while our method uses $N$ when we compute `mean()`. In other words, there is a difference between Sample STD and Population STD.

In [86]:
deviations = returns - returns.mean()
squared_deviations = deviations**2
num_obs = returns.dropna().shape[0]
variance = squared_deviations.sum()/(num_obs - 1)
volatility = np.sqrt(variance)
volatility

Close    0.014428
dtype: float64

In [83]:
returns.std()

Close    0.014428
dtype: float64

We can confirm that now the two values are matched with $0.014428$.

## Annualized Volatility

We can't compare the volatility from daily data with the volatility from monthly data.

$$\large \sigma_{annualized} = \sigma_p \sqrt{p}$$

There are approximately 252 trading days per calendar year.

For example, if the standard deviation of a stock's daily return series is 0.1%. If we annualize it, we get

$$\sigma_{annualized} = 0.001 \times \sqrt{252} = 1.58\%$$

Daily to Annualized: $p=252$

Monthly to Annualized: $p=12$

Quarterly to Annualized: $p=4$

## Sharpe Ratio: Risk Adjusted Measures

We should look at its excess return over the risk free rate.

$$\large \text{Sharpe Ratio = } \frac{R_{p} - R_{f}}{\sigma_{p}} = \frac{\text{RETURN MINUS RISK FREE RATE}}{\text{VOLATILITY}}$$

$$\text{EXCESS RETURN = RETURN OVER THE RISK FREE RATE = RETURN MINUS RISK FREE RATE}$$